### ISSUE TITLE
execute_on = 'ALWAYS' not working as expected with PPs

### ISSUE THREAD

**Issue by GiudGiud** [idaholab/moose#25741]
## Reason
ALWAYS seem to be missing for PPs
It s not that surprising, ALWAYS has to be hardcoded everywhere and we are bound to be missing that code in a few places if we re not careful

## How to reproduce
See discussions below

## Impact
Missing results

### Discussed in https://github.com/idaholab/moose/discussions/25740

<div type='discussions-op-text'>

<sup>Originally posted by **maxnezdyur** October 13, 2023</sup>

When running the MWE below: We should expect both PP to have the same value?
```
Postprocessor Values:
+----------------+----------------+------------------------+
| time           | num_elems_total| num_elems_total_always |
+----------------+----------------+------------------------+
|   0.000000e+00 |   1.600000e+01 |           0.000000e+00 |
|   1.000000e+00 |   1.600000e+01 |           0.000000e+00 |
+----------------+----------------+------------------------+
```
```
[Mesh]
  [mesh]
    type = GeneratedMeshGenerator
    dim = 2
    nx = 4
    ny = 4
    nz = 0
    zmax = 0
    elem_type = QUAD4
  []
[]

[Variables]
  [u]
    order = FIRST
    family = LAGRANGE
  []
[]

[AuxVariables]
  [u_aux]
    order = FIRST
    family = LAGRANGE
  []
  [v_aux]
    order = FIRST
    family = LAGRANGE
  []
[]

[Functions]
  [force]
    type = ParsedFunction
    expression = t
  []
[]

[Kernels]
  [diff]
    type = Diffusion
    variable = u
  []
  [force]
    type = BodyForce
    variable = u
    function = force
  []
[]

[BCs]
  [right]
    type = DirichletBC
    variable = u
    boundary = right
    value = 1
  []
  [left]
    type = DirichletBC
    variable = u
    boundary = left
    value = 0
  []
[]

[Executioner]
  type = Transient
  num_steps = 1
  dt = 1
  solve_type = PJFNK
[]

[Postprocessors]
  [num_elems_total_always]
    type = NumElems
    elem_filter = 'TOTAL'
    execute_on = 'ALWAYS'
  []
  [num_elems_total]
    type = NumElems
    elem_filter = 'TOTAL'
    execute_on = 'initial timestep_end'
  []
[]

[Outputs]
  csv = true
[]
```</div>

**Comment by lindsayad** [comment]
This execute option probably isn't working as expected with many systems...

**Comment by GiudGiud** [comment]
Closed as addressed by [#25859](https://github.com/idaholab/moose/pull/25859)
